### DISCUSSION TITLE
FileAttachment API from nodejs

### DISCUSSION THREAD

**Discussion by declann** [observablehq/framework#1455] [Q&A] [answered]
Hi,

Can we use the FileAttachment API from nodejs, in order to easily move logic from `md` into a JS loader?

I have a few Excel file loads, and dropping down to `exceljs` is a little messy relatively. I'm on the verge of copying the handy functions from `@observablehq/stdlib/src/xlsx.js`.

But it would be much nicer if I can just use the FileAttachment API in nodejs.

I tried:

`import {FileAttachment} from "@observablehq/stdlib";`

but `FileAttachment` isn't exported; `FileAttachments` is, but it's not something I've seen used directly anyplace?

Any pointers much appreciated!

Thanks,
Declan

**Comment by mbostock** [comment, marked as answer]
This isn’t currently supported, no. You can use `readFile` from `fs` to read files in Node.js, and you’re free to copy the helper methods for parsing out of our standard library. But, be aware that Framework doesn’t actually use [`@observablehq/stdlib`](https://github.com/observablehq/stdlib) for the standard library; we instead forked it to create a new version for Framework:

https://github.com/observablehq/framework/tree/main/src/client/stdlib

For reading Excel files:

https://github.com/observablehq/framework/blob/main/src/client/stdlib/xlsx.js

I’d like to extract this code from Framework and publish it to npm; please upvote #852 if you are interested in this feature.

**Reply by declann** [reply]
Thanks for the answer and pointers!
